shell/platform/darwin/ios/ios_surface_factory.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_IOS_IOS__SURFACE_FACTORY_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_IOS_IOS__SURFACE_FACTORY_H_
+
+#include <memory>
+
+#import "flutter/shell/platform/darwin/ios/ios_surface.h"
+#import "flutter/shell/platform/darwin/ios/rendering_api_selection.h"
+
+namespace flutter {
+
+class IOSSurfaceFactory {
+ public:
+  static std::shared_ptr<IOSSurfaceFactory> Create(
+      IOSRenderingAPI rendering_api);
+
+  explicit IOSSurfaceFactory(std::shared_ptr<IOSContext> ios_context);
+
+  ~IOSSurfaceFactory();
+
+  void SetPlatformViewsController(
+      FlutterPlatformViewsController* platform_views_controller);
```

Comment:
You aren't suppose to pass objects via raw pointers: https://google.github.io/styleguide/cppguide.html#Ownership_and_Smart_Pointers

Reply:
This style isn't really respected in the rest of the codebase. There are multiple places where `platform_views_controller` is passed as a raw pointer. I looked at changing it to a smart pointer, it's a pretty big refactor.

Reply:
What's the motto on the team, embrace Zach's shave?

The codebase purports to follow Google C++ Style of which this is in violation.  This violate isn't really style like where to put curly braces, but is policy to avoid creating runtime minefields.  I don't even think there is a good justification for this class but if you want to keep it and not use smart pointers you could pass the FlutterPlatformViewsController as an argument to the CreateSurface.  That way you would be avoiding the creation of another mine.